File: 009/Problem9.c
#include <stdio.h>
#include <time.h>
#include <stdbool.h>

void problem() {
    int ans = 0;
    int c = 0;
    int const s = 1000;
    bool found = false;
    for (int a = 1; a < s/3; a++) {
        for (int b = a; b < s/2; b++){
            c = s - a - b;

            if (a * a + b * b == c * c) {
                ans = a * b * c;
                found = true;
                break;
            }
        }
        if (found) {
            break;
        }
    }
    printf("%d\n", ans);
}

int main() {
    clock_t t;
    t = clock();
    for (int i = 0; i < 1000; i++) {
        problem();
    }
    t = clock() - t;
    double time_taken = ((double) t ) / CLOCKS_PER_SEC;
    printf("Function took %f \n", time_taken);
    return 0;
}

File: 009/Problem9.kt
import kotlin.system.measureNanoTime

fun main() {
    val loops = 1000
    val timeInNanos = measureNanoTime {
        for (i in 1..loops) {
            problem()
        }
    }
    println("Function Took $timeInNanos Nano seconds for $loops loops")
}

fun problem() {
    var a = 1
    var b = a
    var c = 0
    val s = 1000
    var found = false
    while (a < s / 3) {

        b = a
        while (b < s / 2) {
            c = s - a - b

            if (a * a + b * b == c * c) {
                found = true
                break
            }

            b++
        }

        if (found) {
            break
        }
        a++
    }
    println(a * b * c)
}